recipes/diligent-tools/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+import os
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+class DiligentToolsConan(ConanFile):
+    name = "diligent-tools"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/DiligentGraphics/DiligentTools/"
+    description = "Diligent Core is a modern cross-platfrom low-level graphics API."
+    license = ("Apache 2.0")
+    topics = ("graphics")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], 
+    "fPIC": [True, False]
+    }
+    default_options = {"shared": False, 
+    "fPIC": True
+    }
+    generators = "cmake_find_package", "cmake"
+    _cmake = None
+    exports_sources = ["CMakeLists.txt", "patches/**", "BuildUtils.cmake"]
+    short_paths = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package_id(self):
+        if self.settings.compiler == "Visual Studio":
+            if "MD" in self.settings.compiler.runtime:
+                self.info.settings.compiler.runtime = "MD/MDd"
+            else:
+                self.info.settings.compiler.runtime = "MT/MTd"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+        if self.options.shared:
+            raise ConanInvalidConfiguration("Can't build diligent tools as shared lib")
+
+    def requirements(self):
+        self.requires("diligent-core/2.5.1")
+        self.requires("imgui/1.86")
+
+    def diligent_platform(self):
```

Comment:
```suggestion
    @property
    def _diligent_platform(self):
```

Any custom method must be protected